### PR TITLE
Adding VRP examples and test + update routing examples

### PR DESCRIPTION
**Pull request by g-poveda**
- Some standalone example script for the vrp problem are implemented
- unit test for ortools and greedy vrp solver implemented
- Also took this PR to update the example of the pickup_vrp problems, closely related to classical VRP, so that they run well with latest change in the library.
- fix LP solver for VRP

### CODE REVIEW

File: discrete_optimization/vrp/solver/lp_vrp_iterative.py
Comment (nhuet) [inline discrete_optimization/vrp/solver/lp_vrp_iterative.py:1]: It is not a big deal, but when you are modifying discrete_optimization/vrp/solver/*, you are not updating/adding tests and examples but rather fixing issues. 
Ideally you should split this commit into code fixing and tests/examples updating. Or you could add a line in your commit about fixing code.

File: examples/pickup_vrp/loading_example.py [hunk 1]
```diff
@@ -230,9 +218,10 @@ def check_solution(res, gpdp: GPDP):
         n_solutions=10000,
     )
     logging.basicConfig(level=logging.DEBUG)
-    results = solver.solve_intern()
```
Comment (nhuet) [inline examples/pickup_vrp/loading_example.py:233]: I think there are other functions in this file that are using the old api `solve_intern()`. As they are not called in `if __name__ == "__main__": `block, the example is still working though. I made the modifications in my branch on ortools solver, so maybe you can let it as is and i will push my own modifications later.

**Review by nhuet — APPROVED** [review]

File: examples/pickup_vrp/loading_example.py (hunk 1 above)
Comment (g-poveda) [reply examples/pickup_vrp/loading_example.py]: done

File: discrete_optimization/vrp/solver/lp_vrp_iterative.py
Comment (g-poveda) [reply discrete_optimization/vrp/solver/lp_vrp_iterative.py:1]: done, thanks !

**Pull request merged**